### PR TITLE
chore: initial project setup

### PR DESCRIPTION
**Pull request by KhudaDad414**
**Description**

- this PR will initialize this repository.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "optimizer",
+  "version": "0.0.1",
+  "description": "This library will optimize the AsyncAPI specification file.",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "tsc"
```
Comment (magicmatatjahu) [inline package.json:8]: You should also add scripts related to automatic workflows:

```suggestion
    "test": "echo \"No test specified\"",
    "build": "tsc",
    "lint": "echo \"Skip linting\"",
    "get:version": "echo $npm_package_version",   
    "get:name": "echo $npm_package_name",
    "generate:assets": "echo \"Skip generation\"",
    "bump:version": "npm --no-git-tag-version --allow-same-version version $VERSION"
```

We need them for workflows https://github.com/asyncapi/.github/blob/master/.github/workflows/if-nodejs-pr-testing.yml and https://github.com/asyncapi/.github/blob/master/.github/workflows/if-nodejs-version-bump.yml

And of course we shouldn't return the `0` flag from not implemented script :)

File: package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "optimizer",
+  "version": "0.0.1",
+  "description": "This library will optimize the AsyncAPI specification file.",
+  "main": "index.js",
```
Comment (magicmatatjahu) [inline package.json:5]: Output will be in `./lib` so you should change it to (and also add types for TS):

```suggestion
  "main": "./lib/index.js",
  "types": "./lib/index.d.ts",
```

File: package.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "tsc"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/asyncapi/optimizer.git"
+  },
+  "author": "",
```
Comment (magicmatatjahu) [inline package.json:14]: You can describe yourself :)

File: .eslintrc [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  "plugins": [
+    "@typescript-eslint",
+    "sonarjs",
+    "security",
+    "github"
+  ],
+  "extends": [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended",
```
Comment (magicmatatjahu) [inline .eslintrc:12]: They add different ruleset? :) It's only a question.

**Review by magicmatatjahu — COMMENTED** [review]
Great job! I know that it's a draft PR but I add some hints :)

File: package.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  "homepage": "https://github.com/asyncapi/optimizer#readme",
+  "devDependencies": {
+    "@semantic-release/commit-analyzer": "^8.0.1",
+    "@semantic-release/github": "^7.2.0",
+    "@semantic-release/npm": "^7.1.1",
+    "@types/jest": "^23.3.2",
+    "@types/node": "^10.11.0",
+    "jest": "^23.6.0",
+    "typescript": "^3.0.3"
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply package.json:27]: Also you should add eslint plugins defined in `.eslintrc` file as dev dependencies

File: .eslintignore [hunk 6]
```diff
@@ -0,0 +1,4 @@
+node_modules
+docs
+lib
+output
```
Comment (magicmatatjahu) [inline .eslintignore:4]: Do we have any `output` folder? :) 

```suggestion
output
```

Comment (magicmatatjahu) [reply .eslintignore:4]: I think that you should add in this file all content/list from `.gitignore` :)

**Comment by magicmatatjahu** [comment]
Also, please update the title of PR to something like `feat: initial project setup` to follow semantic commit names.

File: .eslintrc (hunk 4 above)
Comment (KhudaDad414) [reply .eslintrc:12]: I don't know man. I saw you guys are using these plugins [here](https://github.com/asyncapi/modelina/blob/master/.eslintrc#L11) and I thought: After all... why not? :)

Comment (magicmatatjahu) [reply .eslintrc:12]: TBH I don't remember 😅 Please remove the first one `plugin:@typescript-eslint/eslint-recommended` It should be the same ruleset.

File: package.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  "homepage": "https://github.com/asyncapi/optimizer#readme",
+  "devDependencies": {
+    "@semantic-release/commit-analyzer": "^8.0.1",
+    "@semantic-release/github": "^7.2.3",
+    "@semantic-release/npm": "^7.1.3",
+    "@types/jest": "^26.0.23",
+    "@types/node": "^15.12.1",
+    "jest": "^27.0.4",
+    "typescript": "^4.3.2"
+  }
```
Comment (magicmatatjahu) [inline package.json:34]: We should also add release related config

```suggestion
  },
  "publishConfig": {
    "access": "public"
  },
  "release": {
    "branches": [
      "master"
    ],
    "plugins": [
        [
          "@semantic-release/commit-analyzer",
            {
              "preset": "conventionalcommits"
            }
        ],
        [
          "@semantic-release/release-notes-generator",
          {
            "preset": "conventionalcommits"
          }
        ],
        "@semantic-release/npm",
        "@semantic-release/github"
    ]
  }
```

File: package.json [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  "main": "./lib/index.js",
+  "types": "./lib/index.d.ts",
+  "scripts": {
+    "test": "echo \"No test specified\"",
+    "build": "tsc",
+    "lint": "eslint --max-warnings 0 --config .eslintrc .",
+    "get:version": "echo $npm_package_version",
+    "get:name": "echo $npm_package_name",
+    "generate:assets": "npm run build",
+    "bump:version": "npm --no-git-tag-version --allow-same-version version $VERSION"
```
Comment (magicmatatjahu) [inline package.json:14]:
```suggestion
    "bump:version": "npm --no-git-tag-version --allow-same-version version $VERSION",
    "release": "semantic-release"
```

**Review by magicmatatjahu — COMMENTED** [review]
We should also add release related scripts and config as in other org repos.

File: package.json [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  "bugs": {
+    "url": "https://github.com/asyncapi/optimizer/issues"
+  },
+  "homepage": "https://github.com/asyncapi/optimizer#readme",
+  "devDependencies": {
+    "@semantic-release/commit-analyzer": "^8.0.1",
+    "@semantic-release/github": "^7.2.3",
+    "@semantic-release/npm": "^7.1.3",
+    "@types/jest": "^26.0.23",
+    "@types/node": "^15.12.1",
```
Comment (magicmatatjahu) [inline package.json:32]: Please add packages related to eslint and eslint plugins:

```suggestion
    "@types/node": "^15.12.1",
    "@typescript-eslint/eslint-plugin": "^4.26.0",
    "@typescript-eslint/parser": "^4.26.0",
    "eslint": "^7.28.0",
    "eslint-plugin-github": "^4.1.3",
    "eslint-plugin-security": "^1.4.0",
    "eslint-plugin-sonarjs": "^0.8.0-125",
```

**Comment by magicmatatjahu** [comment]
We should use the `chore:` in PR's title, because we don't have any code which we can use from package :)

**Comment by KhudaDad414** [comment]
@magicmatatjahu I think it's ready to be merged.

**Comment by magicmatatjahu** [comment]
@KhudaDad414 Please install deps to have `package-lock.json` :)

**Pull request merged**